app/assets/javascripts/angular/controllers/topics_ctrl.coffee

```diff
@@ -15,6 +15,15 @@ angular.module "lean-coffee"
       if confirm('Are you sure you want to delete this?')
         topicsService.delete(topic)
 
+    @archive = (topic) =>
+      if confirm('Are you sure you want to archive this?')
+        topicsService.archive(topic)
+
+    @archiveAll = =>
+      if confirm('Are you sure you want to archive all discussed topics?')
+        topicsService.archiveAll()
+        topicsService.reload()
```

Comment:
Aha! I thought I was just missing something, but I'm glad that I could be of some assistance! To be fair I have probed it no more than you, and so I'd also be curious to know whether Angular updates its model with the response from the server. I'm inclined to think it wouldn't, though, as you'd have to return the same representation from the server on every request and that seems a lot for it to presume.